packages/cli-e2e/__tests__/angular.specs.ts

```diff
@@ -13,92 +13,128 @@ import {isSearchRequest} from '../utils/platform';
+      if (isGenericYesNoPrompt(data.toString())) {
+        await answerPrompt(`y${EOL}`, proc);
+      }
+    });
 
-describe('ui', () => {
-  describe('create:angular', () => {
-    let browser: Browser;
-    let processManager: ProcessManager;
-    // TODO: CDX-90: Assign a dynamic port for the search token server on all ui projects
-    const clientPort = '4200';
-    const projectName = `${process.env.GITHUB_ACTION}-angular-project`;
-    const searchPageEndpoint = `http://localhost:${clientPort}`;
-    const tokenProxyEndpoint = `http://localhost:${clientPort}/token`;
-    let interceptedRequests: HTTPRequest[] = [];
-    let page: Page;
-
-    const searchboxSelector = 'app-search-page app-search-box input';
-
-    beforeAll(async () => {
-      processManager = new ProcessManager();
-      browser = await getNewBrowser();
-      const buildProcess = setupUIProject(
-        processManager,
-        'ui:create:angular',
-        projectName,
-        {
-          flags: ['--defaults'],
-        }
-      );
-
-      buildProcess.stdout.on('data', async (data) => {
-        if (isGenericYesNoPrompt(data.toString())) {
-          await answerPrompt(`y${EOL}`, buildProcess);
-        }
-      });
-
-      await Promise.race([
-        new Promise<void>((resolve) => {
-          buildProcess.on('exit', async () => {
-            resolve();
-          });
-        }),
-        new Promise<void>((resolve) => {
-          buildProcess.stdout.on('data', (data) => {
-            if (
-              /Happy hacking !/.test(
-                stripAnsi(data.toString()).replace(/\n/g, '')
-              )
-            ) {
-              resolve();
-            }
-          });
-        }),
-      ]);
-
-      const startServerProcess = processManager.spawn('npm', ['run', 'start'], {
-        cwd: getProjectPath(projectName),
-      });
-
-      await new Promise<void>((resolve) => {
-        startServerProcess.stdout.on('data', async (data) => {
+    return Promise.race([
+      new Promise<void>((resolve) => {
+        proc.on('exit', async () => {
+          resolve();
+        });
+      }),
+      new Promise<void>((resolve) => {
+        proc.stdout.on('data', (data) => {
           if (
-            /Compiled successfully/.test(
+            /Happy hacking !/.test(
               stripAnsi(data.toString()).replace(/\n/g, '')
             )
           ) {
             resolve();
           }
         });
+      }),
+    ]);
+  };
+
+  const startApplication = async (processManager: ProcessManager) => {
+    const proc = processManager.spawn('npm', ['run', 'start'], {
+      cwd: getProjectPath(projectName),
+    });
+
+    return new Promise<void>((resolve) => {
+      proc.stdout.on('data', async (data) => {
+        if (
+          /Compiled successfully/.test(
+            stripAnsi(data.toString()).replace(/\n/g, '')
+          )
+        ) {
+          resolve();
+        }
       });
-    }, 420e3);
```

Comment:
Beware of this one, default timeout is 5 seconds. Ain't sure that's gonna cut it.